python/revng/model/base.py

```diff
@@ -0,0 +1,170 @@
+from dataclasses import dataclass, fields
+from enum import Enum
+from typing import Generic, Type, TypeVar, get_args, get_origin, get_type_hints
+
+import yaml
+
+
+no_default = object()
+
+
+@dataclass
```

Comment:
The other day I was using `@dataclass`, but I find it to have a subtle and undesirable feature: you can set any field you want, even if it was not defined. Is there a way to disable this behavior?

Reply:
That's standard python behavior, not dataclasses. But yes, it should be possible to override `__setattr__` in the base class and deny setting unexpected members

Reply:
This simple trick seems to do the job :)
```
def __setattr__(self, key, value):
    # Prevent setting undefined attributes
    if self.__dataclass_fields__.get(key) is None:
        raise AttributeError(f"Cannot set attribute {key} for class {type(self).__name__}")
    super().__setattr__(key, value)
```